File: TestApi/Model/views/RoleModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using VDS.Security.Permissions;

namespace ApiServer.Model.views
{
    public static class RoleModel
    {
        public class RoleBase
        {
            public int Id { get; set; }
            public string RoleName { get; set; }
            public string Descriptions { get; set; }
        }
        public class RoleForUpdate : RoleBase
        {
            public long LastModifierUserId { get; set; }
        }

        public class RoleForCreate : RoleBase
        {
            public long CreatorUserId { get; set; }
        }
        
        public class RoleForView : RoleBase
        {
            public List<PermissionModel.PermissionWithCategor> Permissions { get; set; }
        }
    }
}

File: Client/src/app/shared/models/role.model.ts
import { IUser } from './user.model';
import { RoleClaim } from './roleClaim.model';
import { Permission, PermissionForView, PermissionCategory } from './permission.model';
export interface IRole {
    id: number
    roleName: string;
    descriptions: string;
}

export class Role implements IRole{
    constructor(public id: number, public roleName: string, public normalizedRoleName: string, public permissions: Permission[],public lastModifierUser: IUser, public deleterUser: IUser, public isDeleted: boolean, public deleterUserId: number, public deletionTime: Date, public lastModificationTime: Date, public lastModifierUserId: number, public creationTime:Date, public creatorUserId:number, public creatorUser: IUser, public descriptions: string){
    }
}

export class RoleForView implements IRole{
    constructor(public id: number, public roleName: string, public descriptions: string, public permissions: PermissionCategory[]){

    }
}

export class RoleForUpdate implements IRole{
    constructor(public id: number, public roleName: string, public descriptions: string, public lastModifierUserId: any){

    }
}

export class RoleForCreate implements IRole{
    constructor(public id: number, public roleName: string, public descriptions: string, public creatorUserId: any){

    }
}